### DISCUSSION TITLE
how to gat user state via id

### DISCUSSION THREAD

**Discussion by glebkhyl** [aiogram/aiogram#1098] [Q&A] [answered]
### aiogram version

2.x

### Problem

How to get the state of a certain person?

### Possible solution

i tried to do so "await state.get_state(user=some telegram id)" , but it didn't work out

**Comment by Forden** [comment]
It's not an issue with aiogram, try joining community groups and asking how to solve your problem there.

**Comment by JrooTJunior** [comment, marked as answer]
The `state: FSMContext` object is a proxy for the FSM Storage interface with an associated user context. To change state of another user, you will need to interract with storage durectly or create an instance of `FSMContext`.

For example:

```python
user_state = dp.current_state(user_id=42, chat_id=42)
user_state.set_state(another_state)
```
